spec/features/changing_answers_spec.rb

```diff
@@ -1,284 +1,226 @@
+    claim = start_student_loans_claim
+    claim.update!(attributes_for(:claim, :submittable))
+    claim.eligibility.update!(attributes_for(:student_loans_eligibility, :eligible))
+    visit claim_path(StudentLoans.routing_name, "check-your-answers")
+
+    new_claim_school = create(:school, :student_loan_eligible, name: "Claim School")
+
+    find("a[href='#{claim_path(StudentLoans.routing_name, "claim-school")}']").click
 
-    expect(find("#eligible_subjects_physics_taught").checked?).to eq(true)
+    choose_school new_claim_school
 
-    uncheck I18n.t("student_loans.questions.eligible_subjects.physics_taught"), visible: false
+    expect(claim.eligibility.reload.claim_school).to eql new_claim_school
+    expect(claim.eligibility.physics_taught).to be_nil
+    expect(claim.eligibility.biology_taught).to be_nil
+    expect(claim.eligibility.employment_status).to be_nil
+    expect(claim.eligibility.current_school).to be_nil
+
+    expect(current_path).to eq(claim_path(StudentLoans.routing_name, "subjects-taught"))
 
     check I18n.t("student_loans.questions.eligible_subjects.biology_taught"), visible: false
     check I18n.t("student_loans.questions.eligible_subjects.chemistry_taught"), visible: false
 
     click_on "Continue"
 
-    expect(eligibility.reload.physics_taught).to eq(false)
-    expect(eligibility.biology_taught).to eq(true)
-    expect(eligibility.chemistry_taught).to eq(true)
+    expect(claim.eligibility.reload.biology_taught).to eq(true)
+    expect(claim.eligibility.chemistry_taught).to eq(true)
+
+    expect(current_path).to eq(claim_path(StudentLoans.routing_name, "still-teaching"))
+
+    choose_still_teaching "Yes, at Claim School"
+
+    expect(claim.eligibility.reload.employment_status).to eql("claim_school")
+    expect(claim.eligibility.current_school).to eql new_claim_school
 
     expect(current_path).to eq(claim_path(StudentLoans.routing_name, "check-your-answers"))
   end
 
-  scenario "Teacher changes their leadership position to no" do
-    claim.eligibility.had_leadership_position = true
-    claim.save!
+  scenario "Teacher changes an answer which is a dependency of some of the subsequent answers they’ve given, making them ineligible" do
+    claim = start_student_loans_claim
+    claim.update!(attributes_for(:claim, :submittable))
+    claim.eligibility.update!(attributes_for(:student_loans_eligibility, :eligible, had_leadership_position: false))
+    visit claim_path(StudentLoans.routing_name, "check-your-answers")
 
     find("a[href='#{claim_path(StudentLoans.routing_name, "leadership-position")}']").click
 
-    choose "No"
+    choose "Yes"
     click_on "Continue"
 
-    expect(eligibility.reload.had_leadership_position).to eq(false)
+    expect(claim.eligibility.reload.had_leadership_position).to eq(true)
+    expect(claim.eligibility.reload.mostly_performed_leadership_duties).to be_nil
```

Comment:
We could save a database call here as the record has already been reloaded
```suggestion
    expect(claim.eligibility.mostly_performed_leadership_duties).to be_nil
```

Reply:
Thanks, good catch. Fixed it.